polarssl/files/polarssl-1.3.9.patch

```diff
@@ -0,0 +1,24 @@
+diff -Naur original/CMakeLists.txt polarssl-1.3.9/CMakeLists.txt
+--- original/CMakeLists.txt	2014-10-20 08:59:19
++++ polarssl-1.3.9/CMakeLists.txt	2024-08-19 09:09:52
+@@ -25,7 +25,7 @@
+ set(CMAKE_BUILD_TYPE ${CMAKE_BUILD_TYPE}
+     CACHE STRING "Choose the type of build: None Debug Release Coverage ASan Check CheckFull"
+     FORCE)
+-
++set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -Wno-error=incompatible-pointer-types")
```

Comment:
What was seen without this present? Without it I don't get any incompatible pointer type errors. Perhaps it's toolchain version specific, or something we've already changed in KOS in the meantime? 

If you still get those errors could you please provide a log of them? Might indicate something that needs fixing inside kos/dc-chain.

Reply:
I'm actually using 15.0.0. I'll see if I can test on 14.2.0, but running it plainly with your patch and removing the cflag override I am able to see the error (I hadn't tried with only the one patch change at a time). 

Removing your whole patch and just passing `-DHAVE_MACHINE_ENDIAN_H` in the cmake args though seems to avoid the error. The error's being generated by the same net.c file, so it's probably that passing the right define sets things up somewhere in cmake so that there's no pointer incompatibility.

Could you please test with the patch disabled and just passing the cmake arg that I noted?

Reply:
Sure, I'll test with your suggestions and report back my results on 14.2.0.

Reply:
I tested adding that flag but as mentioned, CMake is not taking that flag `-DHAVE_MACHINE_ENDIAN_H`.

What I think you experienced, is that for some reason when cmake fails , the KOS script falls back to regular `make`, and somehow continues building, but that's not the expected build:

```
CMake Error: Parse error in command line argument: HAVE_MACHINE_ENDIAN_H
 Should be: VAR:type=value

CMake Error: Run 'cmake --help' for all supported options.
  CC    aes.c
  CC    aesni.c
  CC    arc4.c
  CC    asn1parse.c
.....
```

Also as I wrote in the other comments, if I use flag `-DHAVE_MACHINE_ENDIAN_H=1` then Cmake works and start building but then it fails on `net.c` due to the endian.h include.

Are you sure your cmake actually handled the flag correctly? 
I think the patch is required and the flags need to stay like I proposed in my PR.

Reply:
My apologies, I had missed that and it does seem that for me the fallback manages to build somehow. I had entirely missed the notice being presented regarding ignoring all the options, and I entirely forgot that you can't simply pass in cflags into cmake.

Thanks for working through these for me. Hopefully someone can figure out what's going on with this in the future since somehow it works without the patch if an error gets triggered. In the meantime, this is all just fine as a way to overcome that.